data/decorators.py

```diff
@@ -0,0 +1,4 @@
+def deprecated_model(model_cls):
+    """ Marks a model has deprecated, and ensures no writings occur on it. """
+    model_cls.__deprecated_model = True
+    return model_cls
```

Comment:
This kinda feels weird being in a separate file because it doesn't really do anything unless the code at the bottom of data.database doesn't execute. If anything what you'd want here is a method for checking if a model is deprecated that can be used elsewhere without understanding that the class has a hidden attr as an implementation.

Reply:
Good idea. I'll add that.